src/Components/Onboarding/Steps/Artists/index.tsx

```diff
@@ -22,20 +23,36 @@ const OnboardingSearchBox = styled.div`
 
 interface State {
   inputText: string
+  inputTextQuery: string
 }
 
 export default class Artists extends React.Component<StepProps, State> {
   state = {
     inputText: "",
+    inputTextQuery: "",
   }
 
-  // onInputChange = e => {
-  //   this.props.onStateChange({ nextButtonEnabled: true })
-  // }
+  componentDidMount() {
+    this.throttledTextChange = throttle(this.throttledTextChange, 500, {
+      leading: true,
+    })
```

Comment:
This is a bit of a brain teaser. We could possibly use the [`lodash-decorators`](https://www.npmjs.com/package/lodash-decorators) to make it more declarative: 

```ts
@Bind()
@Debounce(500, { leading: true }) 
throttledTextChange(inputText) {
  this.setState({ inputTextQuery: inputText })
}
```

Reply:
Ooh, this is really cool. I don't have a preference. Should I go for this?